src/libs/tmdk/e2e-tests/basic-e2e/basic-e2e.test.ts

```diff
@@ -0,0 +1,492 @@
+      "s3ArnDelete",
+      workspaceS3BucketArn + "/DO_NOT_DELETE_WORKSPACE_*"
+    );
+    await aws().iam.putRolePolicy({
+      RoleName: twinMakerRoleName,
+      PolicyName: twinMakerPolicyName,
+      PolicyDocument: twinMakerPolicyString,
+    });
+    await delay(10000); // allow role to propagate
+    console.log("Creating workspace bucket: " + workspaceS3BucketName);
+    await aws().s3.createBucket({
+      Bucket: workspaceS3BucketName,
+    });
+    console.log("Creating workspace: " + constants.workspaceId);
+    await aws().tm.createWorkspace({
+      workspaceId: constants.workspaceId,
+      s3Location: workspaceS3BucketArn,
+      role: twinMakerRoleArn,
+    });
+    console.log(
+      "Uploading scene 1 definition file to s3 bucket: " + workspaceS3BucketName
+    );
+    const scene1Definition = JSON.parse(
+      fs.readFileSync(
+        path.join(constants.localResourcesDir, constants.scene1FileName),
+        constants.jsonEncoding
+      )
+    );
+    scene1Definition["nodes"][0]["components"][0]["uri"] = model1S3Location;
+    const scene1UploadParams = {
+      Bucket: workspaceS3BucketName,
+      Key: constants.scene1FileName,
+      Body: JSON.stringify(scene1Definition),
+    };
+    await aws().s3.send(new PutObjectCommand(scene1UploadParams));
+    constants.scene1Input["contentLocation"] = scene1S3Location;
+    console.log(
+      "Uploading model 1 glb file to s3 bucket: " + workspaceS3BucketName
+    );
+    const model1UploadParams = {
+      Bucket: workspaceS3BucketName,
+      Key: constants.model1FileName,
+      Body: fs.createReadStream(
+        path.join(constants.localResourcesDir, constants.model1FileName)
+      ),
+    };
+    await aws().s3.send(new PutObjectCommand(model1UploadParams));
+    console.log("Successfully set up test resources.");
+  } catch (e) {
+    console.error(
+      "Error while setting up test resources, please check logs, clean up, and restart test. \n",
+      e
+    );
+    throw e;
+  }
+
+  // 3. Create some TwinMaker resources
+  console.log("Setting up first round of TwinMaker resources.");
+  try {
+    await aws().tm.createComponentType(constants.componentType1Input);
+    await aws().tm.createScene(constants.scene1Input); // Scene already contains model 1
+    await aws().tm.createEntity(constants.entity1Input);
+    console.log("Successfully set up first round of TwinMaker resources.");
+  } catch (e) {
+    console.error(
+      "Error while creating TwinMaker resources, please check logs, clean up, and restart test. \n",
+      e
+    );
+    throw e;
+  }
+
+  // 4. Init tmdk project
+  console.log(
+    "Using init to initialize tmdk project in dir: " + constants.tmdkDirectory
+  );
+  argv2 = {
+    _: ["init"],
+    $0: "tmdk_local",
+    region: constants.region,
+    "workspace-id": constants.workspaceId,
+    out: constants.tmdkDirectory,
+  } as Arguments<init.Options>;
+  expect(await init.handler(argv2)).toBe(0);
+
+  // 5. Validate tmdk definition
```

Comment:
would it be simpler to compare against a pre-saved tmdk definition?

Reply:
Yes, will add